test/e2e/far_e2e_test.go

```diff
@@ -75,33 +73,53 @@ var _ = Describe("FAR E2e", func() {
 
 	Context("stress cluster", func() {
 		var (
-			nodeName           string
-			nodeBootTimeBefore time.Time
-			err                error
+			nodes, filteredNodes *corev1.NodeList
+			nodeName             string
+			nodeBootTimeBefore   time.Time
+			err                  error
 		)
 		BeforeEach(func() {
-			nodes := &corev1.NodeList{}
+			nodes = &corev1.NodeList{}
 			selector := labels.NewSelector()
 			requirement, _ := labels.NewRequirement(medik8sLabels.WorkerRole, selection.Exists, []string{})
 			selector = selector.Add(*requirement)
 			Expect(k8sClient.List(context.Background(), nodes, &client.ListOptions{LabelSelector: selector})).ToNot(HaveOccurred())
 			if len(nodes.Items) < 1 {
 				Fail("No worker nodes found in the cluster")
 			}
-
-			nodeName = randomizeWorkerNode(nodes)
-			previousNodeName = nodeName
+			if filteredNodes != nil {
+				nodes = filteredNodes
+			}
+			selectedNode := randomizeWorkerNode(nodes)
+			nodeName = selectedNode.Name
 			nodeNameParam := v1alpha1.NodeName(nodeName)
 			parameterName := v1alpha1.ParameterName(nodeIdentifierPrefix)
 			testNodeID := testNodeParam[parameterName][nodeNameParam]
 			log.Info("Testing Node", "Node name", nodeName, "Node ID", testNodeID)
+			// filter the last remediated node from the list of available nodes
+			filteredNodes = &corev1.NodeList{}
+			for _, node := range nodes.Items {
+				if node.Name != nodeName {
+					filteredNodes.Items = append(filteredNodes.Items, node)
+				}
+			}
 
 			// save the node's boot time prior to the fence agent call
 			nodeBootTimeBefore, err = e2eUtils.GetBootTime(clientSet, nodeName, testNsName, log)
 			Expect(err).ToNot(HaveOccurred(), "failed to get boot time of the node")
+
+			// set the node as "unhealthy" by disabling kubelet
+			makeNodeUnready(selectedNode)
+
 			far := createFAR(nodeName, fenceAgent, testShareParam, testNodeParam)
 			DeferCleanup(deleteFAR, far)
 		})
+		When("running FAR to reboot one node", func() {
```

Comment:
Probably caused by rebasing etc., but I think this `When()` block can be removed. We test 2 remediations in the `When()` block below, that's enough I think :)